### PR TITLE
PVP 289 Added feedback form

### PR DESCRIPTION
**Pull request by vytjar**
- Created Feedback model and implemented FeedbackService to upload feedback entries to the database
- Created feedback form and feedback dialog that contains the form
- Users can choose to report bug or request a feature, fill description and rate the app from 0 stars to 5 stars
- Feedback dialog can be opened from the drawer upon pressing clickable text on the bottom left of the sign out button

This is how it looks like:

image

image

### CODE REVIEW

File: app/src/main/java/com/pvp/app/ui/screen/feedback/FeedbackForm.kt [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+@Composable
+fun FeedbackForm(
+    model: FeedbackViewModel = hiltViewModel(),
+    onCreate: () -> Unit
+) {
+    var bug by remember { mutableStateOf(true) }
+    var description by remember { mutableStateOf("") }
+    var rating by remember { mutableStateOf(0) }
+
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/feedback/FeedbackForm.kt:59]: Possibly unneeded empty line

File: app/src/main/java/com/pvp/app/ui/screen/feedback/FeedbackForm.kt [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    val isFormValid by remember { derivedStateOf { description.isNotEmpty() } }
+    val localeCreate = stringResource(R.string.action_create)
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/feedback/FeedbackForm.kt:61]: If it is used only in one place, could drop the variable and call stringResource in parameter spot, like other ones.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -181,6 +185,7 @@
     <string name="input_field_activity_label">Activity</string>
     <string name="input_field_date_label">Date</string>
     <string name="input_field_description_edit_label">Editing description</string>
+    <string name="input_field_description_error_cannot_be_empty">Description can not be empty</string>
```
Comment (klaidoshka) [inline app/src/main/res/values/strings.xml:188]: `cannot` maybe?

File: app/src/main/res/values-lt/strings.xml [hunk 4]
```diff
@@ -85,6 +85,10 @@
     <string name="drawer_button_sign_out">Atsijungti</string>
     <string name="drawer_button_sign_out_confirmation">Ar tikrai norite atsijungti?</string>
     <string name="drawer_button_sign_out_error">Nepavyko atsijungti, įvyko klaida</string>
+    <string name="drawer_button_feedback">Palikti atsiliepimą</string>
+    <string name="feedback_form_bug">Klaida</string>
+    <string name="feedback_form_feature_request">Funkcijos užklausa</string>
```
Comment (klaidoshka) [inline app/src/main/res/values-lt/strings.xml:90]: `užklausa` -> `prašymas` maybe?

**Review by AurisTFG — APPROVED** [review]

**Pull request merged**